flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractJobReconciler.java

```diff
@@ -74,6 +77,11 @@ public abstract class AbstractJobReconciler<
 
     public static final String LAST_STATE_DUMMY_SP_PATH = "KUBERNETES_OPERATOR_LAST_STATE";
 
+    private final SnapshotTriggerTimestampStore checkpointTriggerTimestamps =
+            new SnapshotTriggerTimestampStore(CHECKPOINT);
+    private final SnapshotTriggerTimestampStore savepointTriggerTimestamps =
+            new SnapshotTriggerTimestampStore(SAVEPOINT);
```

Comment:
Feels like this should be a single object to wrap these two and hide the details